registries/adapters/galaxy_adapter.go

```diff
@@ -0,0 +1,259 @@
+	Spec bundle.Spec `json:"apb_metadata"`
+}
+
+// GalaxyRoleSummary - Role Summary obtained from Role Response.
+type GalaxyRoleSummary struct {
+	Namespace GalaxyRoleNamespace `json:"namespace"`
+}
+
+// GalaxyRoleNamespace - Role Namespace obtained from Role Response Summary.
+type GalaxyRoleNamespace struct {
+	Name string `json:"name"`
+}
+
+// GalaxySearchResponse - Search response for Galaxy.
+type GalaxySearchResponse struct {
+	Count   int           `json:"count"`
+	Results []*GalaxyRole `json:"results"`
+	Next    string        `json:"next"`
+}
+
+// RegistryName - Retrieve the registry name
+func (r GalaxyAdapter) RegistryName() string {
+	return galaxyName
+}
+
+// GetImageNames - retrieve the images
+func (r GalaxyAdapter) GetImageNames() ([]string, error) {
+	log.Debug("GalaxyAdapter::GetImages")
+	log.Debug("BundleSpecLabel: %s", BundleSpecLabel)
+	log.Debug("Loading role list with tag: [apb]")
+
+	channel := make(chan string)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	var imageResp *GalaxySearchResponse
+	var err error
+	// Intial call to getNextImages this will fan out to retrieve all the values.
+	if len(r.Config.Org) == 0 {
+		imageResp, err = r.getNextImages(ctx, galaxySearchURL, channel, cancelFunc)
+	} else {
+		imageResp, err = r.getNextImages(ctx, fmt.Sprintf(galaxyNSSearchURL, r.Config.Org), channel, cancelFunc)
+	}
+	// if there was an issue with the first call, return the error
+	if err != nil {
+		return nil, err
+	}
+
+	// If no results in the fist call then close the channel as nothing will get loaded.
+	if len(imageResp.Results) == 0 {
+		log.Info("canceled retrieval as no items in org")
+		close(channel)
+	}
+	var apbData []string
+	counter := 1
+	for imageData := range channel {
+		apbData = append(apbData, imageData)
+		if counter < imageResp.Count {
+			counter++
+		} else {
+			close(channel)
+		}
+	}
+	// check to see if the context had an error
+	if ctx.Err() != nil {
+		log.Errorf("encountered an error while loading images, we may not have all the apb in the catalog - %v", ctx.Err())
+		return apbData, ctx.Err()
+	}
+
+	return apbData, nil
+}
+
+// FetchSpecs - retrieve the spec for the image names.
+func (r GalaxyAdapter) FetchSpecs(imageNames []string) ([]*bundle.Spec, error) {
+	specs := []*bundle.Spec{}
+	for _, imageName := range imageNames {
+		spec, err := r.loadSpec(imageName)
+		if err != nil {
+			log.Errorf("Failed to retrieve spec data for image %s - %v", imageName, err)
+		}
+		if spec != nil {
+			specs = append(specs, spec)
+		}
+	}
+	return specs, nil
+}
+
+// getNextImages - will follow the next URL using go routines.
+func (r GalaxyAdapter) getNextImages(ctx context.Context,
+	url string, ch chan<- string,
+	cancelFunc context.CancelFunc) (*GalaxySearchResponse, error) {
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		log.Errorf("unable to get next roles for url: %v - %v", url, err)
+		cancelFunc()
+		close(ch)
```

Comment:
are you supposed to close the channel here? seems like the person calling you should do the closing. That is, you did not create the resource so you shouldn't be destroying it.